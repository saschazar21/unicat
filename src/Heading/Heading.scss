@import '../__styles__/variables';

.wrapper {
  display: block;
  font-family: var(--font-heading);
  margin-top: 0;
  margin-bottom: spacing(xs);
  overflow: hidden;
  white-space: nowrap;
  text-align: inherit;
  text-overflow: ellipsis;

  @include media-query-up(m) {
    margin-bottom: spacing(m);
  }
}

@each $heading in $headings {
  .#{$heading} {
    $i: map-get($numbers, str-slice($heading, -1));
    $oneDown: h#{$i + 1};
    @if $oneDown == h6 {
      font-size: fontSize(h5);
    }
    font-size: fontSize($oneDown);

    @include media-query-up(m) {
      font-size: fontSize($heading);
    }
  }
}
