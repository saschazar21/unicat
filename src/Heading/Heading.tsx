import React, { Component, ReactNode } from 'react';
import classnames from 'classnames';

import styles from './Heading.scss';

export interface HeadingProps {
  children: ReactNode[] | string;
  className?: string;
  level: Heading;
  SEO?: boolean;
}

export default class Element extends Component<HeadingProps> {
  static defaultProps = {
    className: null,
    SEO: false,
  };

  public render() {
    const { children, className: customClassName, level, SEO } = this.props;
    const className = classnames(
      styles.wrapper,
      styles[level],
      customClassName
    );

    return React.createElement(SEO ? level : 'span', { className }, children);
  }
}
