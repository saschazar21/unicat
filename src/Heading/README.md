A simple header text, may be customized using the `level` prop.

```jsx
<Heading level={1}>
  I'm a <code>h1</code> by default
</Heading>
```

When explicitely setting the `SEO` prop to `false`, a `<span>` tag is rendered:

```jsx
<Heading level={3} SEO={false}>
  I'm a span looking like a <code>h3</code>
</Heading>
```
