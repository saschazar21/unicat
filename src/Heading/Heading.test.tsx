import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Heading from './Heading';

describe('Heading', () => {
  it('renders', () => {
    const wrapper = shallow(<Heading level="h1">Test</Heading>);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('span')).toHaveLength(1);
    expect(wrapper.find('h1')).toHaveLength(0);
    expect(wrapper.find('.h1')).toHaveLength(1);
  });

  it('renders a <h2 /> tag', () => {
    const wrapper = shallow(<Heading level="h2" SEO>Test</Heading>);

    expect(toJson(wrapper)).toMatchSnapshot();
    
    expect(wrapper.find('h2')).toHaveLength(1);
    expect(wrapper.find('span')).toHaveLength(0);
  });
})