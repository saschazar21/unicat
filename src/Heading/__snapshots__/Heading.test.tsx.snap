// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders 1`] = `
<span
  className="wrapper h1"
>
  Test
</span>
`;

exports[`Heading renders a <h2 /> tag 1`] = `
<h2
  className="wrapper h2"
>
  Test
</h2>
`;
